classes/PrestaShopAutoload.php

```diff
@@ -172,70 +182,7 @@ public function generateIndex()
             }
         }
 
-        $coreClasses = $this->getClassesFromDir('classes/');
-
-        $classes = array_merge(
-            $coreClasses,
-            $this->getClassesFromDir('controllers/')
-        );
-
-        $contentNamespacedStub = '<?php ' . "\n" . 'namespace PrestaShop\\PrestaShop\\Adapter\\Entity;' . "\n\n";
-
-        foreach ($coreClasses as $coreClassName => $coreClass) {
-            if (substr($coreClassName, -4) == 'Core') {
-                $coreClassName = substr($coreClassName, 0, -4);
-                if ($coreClass['type'] != 'interface') {
-                    $contentNamespacedStub .= $coreClass['type'] . ' ' . $coreClassName . ' extends \\' . $coreClassName . ' {};' . "\n";
-                }
-            }
-        }
-
-        if ($this->_include_override_path) {
-            $coreOverrideClasses = $this->getClassesFromDir('override/classes/');
-            $coreClassesWOOverrides = array_diff_key($coreClasses, $coreOverrideClasses);
-
-            $classes = array_merge(
-                $classes,
-                $coreOverrideClasses,
-                $this->getClassesFromDir('override/controllers/')
-            );
-        } else {
-            $coreClassesWOOverrides = $coreClasses;
-        }
-
-        $contentStub = '<?php' . "\n\n";
-
-        foreach ($coreClassesWOOverrides as $coreClassName => $coreClass) {
-            if (substr($coreClassName, -4) == 'Core') {
-                $coreClassNameNoCore = substr($coreClassName, 0, -4);
-                if ($coreClass['type'] != 'interface') {
-                    $contentStub .= $coreClass['type'] . ' ' . $coreClassNameNoCore . ' extends ' . $coreClassName . ' {};' . "\n";
-                }
-            }
-        }
-
-        ksort($classes);
-        $content = '<?php return ' . var_export($classes, true) . '; ?>';
-
-        // Write classes index on disc to cache it
-        $filename = static::getCacheFileIndex();
-        @mkdir(_PS_CACHE_DIR_, 0777, true);
-
-        if (!$this->dumpFile($filename, $content)) {
-            Tools::error_log('Cannot write temporary file ' . $filename);
-        }
-
-        $stubFilename = static::getStubFileIndex();
-        if (!$this->dumpFile($stubFilename, $contentStub)) {
-            Tools::error_log('Cannot write temporary file ' . $stubFilename);
-        }
-
-        $namespacedStubFilename = static::getNamespacedStubFileIndex();
-        if (!$this->dumpFile($namespacedStubFilename, $contentNamespacedStub)) {
-            Tools::error_log('Cannot write temporary file ' . $namespacedStubFilename);
-        }
-
-        $this->index = $classes;
+        $this->index = $this->classLoader->buildClassIndex($this->_include_override_path);
```

Comment:
Seems like this method doesn't do the same thing it used to It previously outputted the generated content into a dump file.

Now it only creates the content without exporting it, isn't this a major (breaking) change in this class's behavior? Maybe for some reason it's not needed anymore?